python/tank/platform/software_launcher.py

```diff
@@ -336,47 +382,186 @@ def get_standard_plugin_environment(self):
 
         return env
 
-    def is_version_supported(self, version):
+    def get_supported_software(self):
         """
-        Compares the given version string with the :meth:`~minimum_supported_version`
-        and returns a boolean to indicate whether it is supported by the launcher
-        or not.
+        Performs a search for supported software installations.
+
+        This method can be overridden by a subclass to completely override
+        supported software discovery. The typical requirement, however, is for
+        a subclasses to simply implement the ``_scan_software()`` abstract
+        method to return all variations of the software on the local filesystem.
 
-        :param str version: Version string to test, e.g. "2015" or "2017.3.sp4"
-        :returns: True if supported, False otherwise.
+        The default implementation calls ``_scan_software()`` to return
+        a list of ``SoftwareVersion`` objects, then calls ``_is_supported()`` on
+        each to determine the final list of supported executables.
         """
-        if self.minimum_supported_version is None:
-            return True
 
-        if is_version_older(version, self.minimum_supported_version):
+        software_versions = []
+
+        # retrieve a list of SoftwareVersion objects for all executables
+        for sw_version in self._scan_software():
+            (supported, reason) = self._is_supported(sw_version)
+            if supported:
+                self.logger.debug("Accepting %s" % (sw_version,))
+                software_versions.append(sw_version)
+            else:
+                self.logger.debug(
+                    "Rejecting %s. Reason: %s" % (sw_version, reason))
+                continue
+
+        return software_versions
+
+    ##########################################################################################
+    # protected methods
+
+    def _is_supported(self, sw_version):
```

Comment:
I wonder if it's worth it to provide as detailed an interface that can be overriden. Maybe simply overriding the result of `_is_supported` is enough. What say you @manneohrstrom ?